samza-core/src/main/java/org/apache/samza/runtime/ApplicationRunnerMain.java

```diff
@@ -71,4 +73,13 @@ public static void main(String[] args) throws Exception {
         throw new IllegalArgumentException("Unrecognized operation: " + op);
     }
   }
+
+  private static Optional<ExternalContext> buildExternalContext(Config config) {
+    /*
+     * By default, use an empty ExternalContext here. In a custom fork of Samza, this can be implemented to pass
+     * a non-empty ExternalContext. Only config should be used to build the external context. In the future, components
+     * like the application descriptor may not be available.
```

Comment:
Did you mean "may be available"? I don't think we need to mention that, it's also weird to have the descriptors (or any app logic) be available for this, since this is expected to be external to the application.

Reply:
Actually I'm not sure if we need the external context here, since there's no application processing (and init) going on here - it's only planning.

Reply:
This was related to the future plan of only doing planning once, and then passing a serialized version of the app around. I thought that the app descriptor was only needed for planning, and once the serialization happened, it would not be available. I wanted to clarify in the comment that external context should not be built with the app descriptor in case it is not available in the future. Is my understanding correct?
I agree that it might be odd to use app logic to build this, but you never know, so I added a note.
If ApplicationRunnerMain gets used to launch the LocalApplicationRunner, then I thought it might be needed.

Reply:
No, ApplicationRunnerMain is only for YARN application. Standalone users have thier own "main" (that of the application container), so they don't need this class.

Reply:
Oh ok, since there was an ApplicationRunner interface and we classloaded the concrete ApplicationRunner, I thought that ApplicationRunnerMain could potentially run any app runner. I can remove this. I guess if someone is forking Samza already and needs external context here, then they could just wire that in on their own.